cvmfs/s3fanout.cc

```diff
@@ -244,29 +245,48 @@ void *S3FanoutManager::MainUpload(void *data) {
   LogCvmfs(kLogS3Fanout, kLogDebug, "Upload I/O thread started");
   S3FanoutManager *s3fanout_mgr = static_cast<S3FanoutManager *>(data);
 
+  s3fanout_mgr->InitPipeWatchFds();
+
   // Don't schedule more jobs into the multi handle than the maximum number of
   // parallel connections.  This should prevent starvation and thus a timeout
   // of the authorization header (CVM-1339).
   unsigned jobs_in_flight = 0;
 
-  while (s3fanout_mgr->thread_upload_run_) {
-    JobInfo *info = NULL;
-    {
-      MutexLockGuard m(s3fanout_mgr->jobs_todo_lock_);
-      if (!s3fanout_mgr->jobs_todo_.empty() &&
-          (jobs_in_flight < s3fanout_mgr->pool_max_handles_)) {
-        info = s3fanout_mgr->jobs_todo_.back();
-        s3fanout_mgr->jobs_todo_.pop_back();
+  while (true) {
+    // Check events with 100ms timeout
+    int timeout_ms = 100;
+    int retval = poll(s3fanout_mgr->watch_fds_, s3fanout_mgr->watch_fds_inuse_,
```

Comment:
In the client, we saw a performance drop between 100ms timeout and 1ms timeout.  Can you check whether this makes a difference for uploading?

Reply:
I am not sure which performance drop do you mean. For every sent request, a response should arrive sooner than 100 ms after. The `poll` function should therefore block the thread very rarely. On the other hand, a 1ms timeout is very small and borders with ordinary busy waiting. Anyway, the s3benchmark did not show any noticeable difference.

Reply:
Ok, thanks for cross checking.